deployment/ccip/changeset/aptos/cs_add_token_pool.go

```diff
@@ -0,0 +1,208 @@
+	}
+	// Validate if token address is provided if pool address is specified
+	if cfg.TokenObjAddress == (aptos.AccountAddress{}) && cfg.TokenPoolAddress != (aptos.AccountAddress{}) {
+		return errors.New("token object address must be provided if token pool address is specified")
+	}
+	// No token pool address provided, so no need to validate token address
+	if cfg.TokenObjAddress == (aptos.AccountAddress{}) && cfg.TokenPoolAddress == (aptos.AccountAddress{}) {
+		return nil
+	}
+	// Validate if token already exists with different pool address
+	for token, pool := range state.AptosChains[cfg.ChainSelector].AptosManagedTokenPools {
+		if (token == cfg.TokenAddress) && (pool != cfg.TokenPoolAddress) {
+			return fmt.Errorf("token %s already exists with a different pool address %s", token, pool)
+		}
+		if (pool == cfg.TokenPoolAddress) && (token != cfg.TokenAddress) {
+			return fmt.Errorf("pool %s already exists with a different token address %s", pool, token)
+		}
+	}
+	for token, pool := range state.AptosChains[cfg.ChainSelector].BurnMintTokenPools {
+		if (token == cfg.TokenAddress) && (pool != cfg.TokenPoolAddress) {
+			return fmt.Errorf("token %s already exists with a different pool address %s", token, pool)
+		}
+		if (pool == cfg.TokenPoolAddress) && (token != cfg.TokenAddress) {
+			return fmt.Errorf("pool %s already exists with a different token address %s", pool, token)
+		}
+	}
+	for token, pool := range state.AptosChains[cfg.ChainSelector].LockReleaseTokenPools {
+		if (token == cfg.TokenAddress) && (pool != cfg.TokenPoolAddress) {
+			return fmt.Errorf("token %s already exists with a different pool address %s", token, pool)
+		}
+		if (pool == cfg.TokenPoolAddress) && (token != cfg.TokenAddress) {
+			return fmt.Errorf("pool %s already exists with a different token address %s", pool, token)
+		}
+	}
+	return nil
+}
+
+func (cs AddTokenPool) Apply(env cldf.Environment, cfg config.AddTokenPoolConfig) (cldf.ChangesetOutput, error) {
+	state, err := stateview.LoadOnchainState(env)
+	if err != nil {
+		return cldf.ChangesetOutput{}, fmt.Errorf("failed to load Aptos onchain state: %w", err)
+	}
+
+	aptosChain := env.BlockChains.AptosChains()[cfg.ChainSelector]
+	ab := cldf.NewMemoryAddressBook()
+	seqReports := make([]operations.Report[any, any], 0)
+	proposals := make([]mcms.TimelockProposal, 0)
+	mcmsOperations := []mcmstypes.BatchOperation{}
+
+	deps := operation.AptosDeps{
+		AB:               ab,
+		AptosChain:       aptosChain,
+		CCIPOnChainState: state,
+	}
+
+	// Deploy Aptos Token
+	tokenObjectAddress := cfg.TokenObjAddress
+	tokenAddress := cfg.TokenAddress
+	tokenOwnerAddress := aptos.AccountAddress{}
+	if cfg.TokenObjAddress == (aptos.AccountAddress{}) {
+		deployTokenIn := seq.DeployTokenSeqInput{
+			TokenParams: cfg.TokenParams,
+			MCMSAddress: state.AptosChains[cfg.ChainSelector].MCMSAddress,
+		}
+		deploySeq, err := operations.ExecuteSequence(env.OperationsBundle, seq.DeployAptosTokenSequence, deps, deployTokenIn)
```

Comment:
small nits:
- we could add some nil checks since we are getting the existing data from states
- The sequences run & the output is tightly tied & assumes that we always run these with MCMS, I personally think we should have option to skip MCMS proposal all the way if we wanted to (for local runs/deployments & such)